pandas/core/arrays/categorical.py

```diff
@@ -2035,16 +2035,24 @@ def mode(self, dropna=True):
     # ------------------------------------------------------------------
     # ExtensionArray Interface
 
-    def unique(self):
+    def unique(self, remove_unused_categories: bool = True) -> "Categorical":
```

Comment:
id rather just change/fix the behavior than add a new keyword

Reply:
I`d be +1 on that, but I`ll wait on input from others. One consideration is that groupby with categoricals used the current behaviour. Changing groupby behaviour without deprecation is not a good idea, IMO.

I would like to change the behaviour of Categorical.unique, but keep the current behaviour of groupbys. Then we can seperately discuss how the groupby behaviour can be properly deprecated and changed.